### PR TITLE
[PATCH v4] API: add new crypto algorithms

### PR DESCRIPTION
**Pull request by JannePeltonen**
Add new crypto algorithms in the API. This PR does not add real validation tests for the algorithms but just makes the validation test aware of the new algorithms.

### CODE REVIEW

File: test/validation/api/crypto/util.c [hunk 1]
```diff
@@ -102,6 +124,20 @@ const char *cipher_alg_name(odp_cipher_alg_t cipher)
...
 	case ODP_CIPHER_ALG_ZUC_EEA3:
 		return "ODP_CIPHER_ALG_ZUC_EEA3";
+	case ODP_CIPHER_ALG_SNOW_V:
+		return "ODP_CIPHER_ALG_SNOW_V";
+	case ODP_CIPHER_ALG_SNOW_V_GCM:
+		return "ODP_CIPHER_ALG_SNOW_V_GCM";
+	case ODP_CIPHER_ALG_SM4_ECB:
+		return "ODP_CIPHER_ALG_SM4_ECB";
+	case ODP_CIPHER_ALG_SM4_CBC:
+		return "ODP_CIPHER_ALG_CM4_CBC";
```
Comment (psavol) [inline test/validation/api/crypto/util.c:134]: Typo: _CM4_CBC -> _SM4_CBC

**Comment by JannePeltonen** [comment]
v2:
- fixed typos
- improved SNOW-V-GMAC and SM4-GMAC documentation slightly
- improved auth_range documentation to refer to GMAC algorithms in general
- rebased

**Review by psavol — APPROVED** [review]

**Comment by anoobj** [comment]
The change looks good to me.

Acked-by: Anoob Joseph <anoobj@marvell.com>

**Comment by anoobj** [comment]
The change looks good to me.

Acked-by: Anoob Joseph <anoobj@marvell.com>

**Comment by JannePeltonen** [comment]
v3: added reviewed-by and acked-by tags

**Pull request merged**